webapp/src/components/error.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+
+interface IErrorBoundaryState {
+  hasError: boolean;
+  error: Error | null;
+  info: React.ErrorInfo | null;
+}
+
+interface IErrorBoundaryProps {
+  children: React.ReactNode;
+}
+
+export class ErrorBoundary extends React.Component<IErrorBoundaryProps, IErrorBoundaryState> {
```

Comment:
I think we should change this to functional component to stay consistent with the rest of the components

Reply:
Because function components don't provide 'componentDidCatch', so here I am using class components directly without introducing any third-party libraries to implement it.